src/sourmash/sig/__main__.py

```diff
@@ -916,6 +917,208 @@ def export(args):
+
+        for record in screed.open(filename):
+            seq_mh = query_mh.copy_and_clear()
+
+            # protein? dna?
+            if is_protein:
+                seq_mh.add_protein(record.sequence)
+            else:
+                try:
+                    seq_mh.add_sequence(record.sequence,
+                                        not args.check_sequence)
+                except ValueError as exc:
+                    seqname = record.name
+                    if len(seqname) > 40:
+                        seqname = seqname[:37] + '...'
+                    notify(f"ERROR in sequence '{seqname}', file '{filename}'")
+                    notify(str(exc))
+                    if args.force:
+                        notify("(continuing)")
+                        continue
+                    else:
+                        sys.exit(-1)
+
+            if seq_mh.intersection(query_mh):
+                # match!
+
+                # output matching sequences:
+                if save_seqs:
+                    save_seqs.fp.write(f">{record.name}\n{record.sequence}\n")
+                    n_sequences_found += 1
+                    n_bp_saved += len(record.sequence)
+
+                # output matching k-mers:
+                if kmer_w:
+                    seq = record.sequence
+                    kh_iter = seq_mh.kmers_and_hashes(seq, force=False,
+                                                      is_protein=is_protein)
+                    for kmer, hashval in kh_iter:
+                        if hashval in query_mh.hashes:
+                            found_mh.add_hash(hashval)
+                            n_kmers_found += 1
+                            d = dict(sequence_file=filename,
+                                     sequence_name=record.name,
+                                     kmer=kmer, hashval=hashval)
+                            kmer_w.writerow(d)
+
+                # add seq_mh to found_mh
+                found_mh += seq_mh.intersection(query_mh)
+
+            # provide progress indicator based on bp...
+            n_sequences_searched += 1
+            n_bp_searched += len(record.sequence)
+
+            if n_bp_searched >= progress_threshold:
+                notify(f"... searched {n_bp_searched} from {n_files_searched} files so far")
+                while n_bp_searched >= progress_threshold:
+                    progress_threshold += progress_interval
+
+    # END major for loop. Now, clean up!
+    if save_kmers:
+        save_kmers.close()
+
+    if save_seqs:
+        save_seqs.close()
+
+    if not n_sequences_searched:
+        notify("ERROR: no sequences searched!?")
+        sys.exit(-1)
+
+    # ...and report!
+    notify("DONE.")
+    notify(f"searched {n_sequences_searched} sequences from {n_files_searched} files, containing a total of {n_bp_searched/1e6:.1f} Mbp.")
+
+    if save_seqs:
+        notify(f"matched and saved a total of {n_sequences_found} sequences with {n_bp_saved/1e6:.1f} Mbp.")
```

Comment:
```suggestion
    # ...and report!
    def len_label(no_bp):
        n = 0
        formatFloat = lambda n: f"{n:.1f}" if n%1 else int(n)
        mag_labels = {0 : '', 1: 'K', 2: 'M', 3: 'G', 4: 'T'}
        while no_bp > 2**10: no_bp /= 2**10; n += 1
        return f"{formatFloat(no_bp)} {mag_labels[n]}bp"
        
    notify("DONE.")
    notify(f"searched {n_sequences_searched} sequences from {n_files_searched} files, containing a total of {len_label(n_bp_searched)}.")

    if save_seqs:
        notify(f"matched and saved a total of {n_sequences_found} sequences with {len_label(n_bp_saved)}.")
```

Reply:
This is not a really important suggestion, but it will give a more friendly label.

Reply:
fine, you guilted me into it! See 2b4d7d31 and 0f0b0d016f4 😆